contracts/src/v0.8/l2ep/dev/zksync/ZKSyncSequencerUptimeFeed.sol

```diff
@@ -0,0 +1,236 @@
+import {TypeAndVersionInterface} from "../../../interfaces/TypeAndVersionInterface.sol";
+import {ZKSyncSequencerUptimeFeedInterface} from "./../interfaces/ZKSyncSequencerUptimeFeedInterface.sol";
+import {SimpleReadAccessController} from "../../../shared/access/SimpleReadAccessController.sol";
+import {AddressAliasHelper} from "@zksync/contracts/l2-contracts/contracts/vendor/AddressAliasHelper.sol";
+
+/// @title ZKSyncSequencerUptimeFeed - L2 sequencer uptime status aggregator
+/// @notice L2 contract that receives status updates from a specific L1 address,
+///  records a new answer if the status changed
+contract ZKSyncSequencerUptimeFeed is
+  AggregatorV2V3Interface,
+  ZKSyncSequencerUptimeFeedInterface,
+  TypeAndVersionInterface,
+  SimpleReadAccessController
+{
+  /// @dev Round info (for uptime history)
+  struct Round {
+    uint64 startedAt; // ─╮ The timestamp at which the round started
+    uint64 updatedAt; //  │ The timestamp at which the round was updated
+    bool status; // ──────╯ The sequencer status for the round
+  }
+
+  /// @dev Packed state struct to save sloads
+  struct FeedState {
+    uint80 latestRoundId; // ─╮ The ID of the latest round
+    uint64 startedAt; //      │ The date at which the latest round started
+    uint64 updatedAt; //      │ The date at which the latest round was updated
+    bool latestStatus; // ────╯ The status of the latest round
+  }
+
+  /// @notice Sender is not the L2 messenger
+  error InvalidSender();
+  /// @notice Replacement for AggregatorV3Interface "No data present"
+  error NoDataPresent();
+
+  event L1SenderTransferred(address indexed from, address indexed to);
+  /// @dev Emitted when an `updateStatus` call is ignored due to unchanged status or stale timestamp
+  event UpdateIgnored(bool latestStatus, uint64 latestTimestamp, bool incomingStatus, uint64 incomingTimestamp);
+  /// @dev Emitted when a updateStatus is called without the status changing
+  event RoundUpdated(int256 status, uint64 updatedAt);
+
+  // solhint-disable-next-line chainlink-solidity/all-caps-constant-storage-variables
+  uint8 public constant override decimals = 0;
+  // solhint-disable-next-line chainlink-solidity/all-caps-constant-storage-variables
+  string public constant override description = "L2 Sequencer Uptime Status Feed";
+  // solhint-disable-next-line chainlink-solidity/all-caps-constant-storage-variables
+  uint256 public constant override version = 1;
+
+  /// @dev L1 address
+  address private s_l1Sender;
+  /// @dev s_latestRoundId == 0 means this contract is uninitialized.
+  FeedState private s_feedState = FeedState({latestRoundId: 0, latestStatus: false, startedAt: 0, updatedAt: 0});
+  mapping(uint80 => Round) private s_rounds;
+
+  /// @param l1SenderAddress Address of the L1 contract that is permissioned to call this contract
+  /// @param initialStatus The initial status of the feed
+  constructor(address l1SenderAddress, bool initialStatus) {
+    _setL1Sender(l1SenderAddress);
```

Comment:
I think it's fair to add it. Updated

Reply:
We need to revert this change because both the validator and sequencer uptime feed contracts need each others reference. the idea is for the sequencer to be deployed first, so it needs to start with a zero address, in order to later be updated once the second is deployed